### ISSUE TITLE
onError while using GenericTypeIndicator

### ISSUE THREAD

**Issue by shivam0x** [nmoskalenko/rxFirebase#14]
Getting error while fetching records using GenericTypeIndicator, it was working in the first try.

Error : java.lang.RuntimeException: com.kelvinapps.rxfirebase.exceptions.RxFirebaseDataCastException: unable to cast firebase data response to generic type

Using the very same code in another class and its working there. Same code copy pasted from here itself.

`RxFirebaseDatabase.observeValueEvent(databaseReference,
                DataSnapshotMapper.of(new GenericTypeIndicator<HashMap<String, Vehicle>>() {
                }))
                .subscribe(new Subscriber<HashMap<String, Vehicle>>() {
                    @Override
                    public void onCompleted() {
                        Logger.log(TAG, "onCompleted: ");
                    }

```
                @Override
                public void onError(Throwable e) {
                    Logger.log(TAG, "onError: " + e);
                   // Here I'm getting this
                  // java.lang.RuntimeException: com.kelvinapps.rxfirebase.exceptions.RxFirebaseDataCastException: unable to cast firebase data response to generic type
                }

                @Override
                public void onNext(HashMap<String, Vehicle> vehicles) {

                }`
```

**Comment by shivam0x** [comment]
I've resolved this issue. By mistake I was passing wrong dbreference.
But I'm getting same issue when there is no item in the database or that child does not exists. Even then it is going into on error while it should give me HashMap with size 0.

Actually firstly we are showing him his added vehicles and then an option to add vehicles. So it is going into onError for first time itself and my whole login of onNext is getting failed.

Can you help me with this issue.

**Comment by renanferrari** [comment]
This seems related to #9.

**Comment by nmoskalenko** [comment]
Fixed in 0.13, please check
